src/Categories/Functor/Slice.agda

```diff
@@ -96,3 +102,58 @@ module _ {A : Obj} where
       ; F-resp-≈ = λ f≈g → ⟨⟩-cong₂ refl (∘-resp-≈ˡ f≈g)
       }
 
+  -- This can and probably should be restricted
+  -- e.g. we only need exponential objects with A as domain
+  -- I don't think we need all products but I don't have a clear idea of what products we do need
+  module _ (ccc : CartesianClosed) (pullback : ∀ {X} {Y} {Z} (h : X ⇒ Z) (i : Y ⇒ Z) → Pullback h i) where
+
+    open CartesianClosed ccc
+    open Cartesian cartesian
+    open Terminal terminal
+    open BinaryProducts products
+
+    -- Needs better name!
```

Comment:
Yes, it does. Ask on the category theory Zulip? It feels like it ought to have a name. (Does the 1lab have this yet?)

Reply:
1lab calls this functor [`exponentiable→product`](https://1lab.dev/Cat.Diagram.Exponential.html#12821), and what we call `Free` they call [`constant-family`](https://1lab.dev/Cat.Instances.Slice.html#22029). I can't see a counterpart to our `Forgetful` but I've asked in their Discord.

I would like to, following _Polynomial Functors and Polynomial Monads_ (the paper I'm slowly working towards formalize here), call `Forgetful` `Σ`, `Free` `Δ`, and `Coforgetful` `Π`, but those names are even _less_ indicative of their function than what we currently have. _Aspects of Topoi_ uses the same names for `Forgetful` and `Coforgetful`, but calls `Free` `×`.

Reply:
Yes, those single-letter names are even worse. The 1lab's names don't seem so bad?

I could try to summon @TOTBWF to see if he has opinions.

Reply:
I have been SUMMONED ✨🧊 🧙✨ 

To get a good idea for names, it's good to understand exactly what is going on with slices. The intuition here is that a slice `f : X -> I` is how we can encode an I-indexed family of objects X_i without any sort of classifying object like `Type`. In places like `Set`, we can move between the two by taking the fibres of `f`, but in a general category we cannot do this. 

As such, base change `pullback-functorial` should really be thought of as a functor on _slices_ `C/B -> C/A` for a morphism `A -> B`: from the perspective of families, this restricts a family `f : X -> B` to a new family `P -> A` along `f` potentially duplicating or deleting fibres. In other words, we take the fibre product: IE, the pullback. By composing with the projection out of the slice, we've essentially "forgotten" the indexing, and are just looking at the total space of the family.

From this perspective, `Free` isn't really the free slice, but rather _the constant family_ on `I`: each fibre over is a copy of `X`. Furthermore, `Forgetful` doesn't really "forget" a slice: it returns the total space of a family by forgetting the indexing.

`Coforgetful` is the trickiest: the bottom leg of the pullback is an obfuscation of the identity morphism, and the right leg is more or less postcomposition by the projection out of the family `X -> I`. Therefore, this pullback consists of sections of the family `X -> I` internalized as a exponential objects.

With all that background out of the way, here are my suggestions:
- Make `pullback-functorial` a functor on slices, and rename it to `Base-change` or `Reindex`
- `Free` becomes `Constant-family`
- `Forgteful` becomes `Total-space` or `Total`
- `Coforgetful` becomes `Internal-sections` or `Sections`

PS: this is some really great work!

Reply:
I like these names a lot! Thanks for the explanation.

While switching to them, I noticed that `pullback-functorial` (after @TOTBWF's suggestion to make it less forgetful) looks a lot like `Categories.Functor.Slice.BaseChange.BaseChange*`. I haven't quite been able to convince myself that they _are_ the same thing, because there's a lot of redirection and it's a little confusing to get my head around, but if they are, I'd like to remove `pullback-functorial` entirely.